lineman/app/services/thread_query_service.coffee

```diff
@@ -52,15 +43,19 @@ angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
       parts = options.split ' '
       moment().startOf('day').subtract(parseInt(parts[0]), parts[1])
 
-    createFilter = (filter = 'show_all') ->
-      (viewData) ->
-        _.filter viewData, (thread) ->
-          return false if thread.isMuted() and filter != 'show_muted'
-          return false if thread.readerNotLoaded()
-          switch filter
-            when 'show_all'           then true
-            when 'show_muted'         then thread.isMuted()
-            when 'show_unread'        then thread.isUnread()
-            when 'show_participating' then thread.isParticipating()
-            when 'show_starred'       then thread.isStarred()
-            when 'show_proposals'     then thread.hasActiveProposal()
+    applyFilters = (view, filters, queryType) ->
```

Comment:
I think that's right, and I think that might end up resolving our pesky
'tangential' issues as well. Actually I think it just solves everything,
heh.

On Thu, Aug 13, 2015 at 12:56 PM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/services/thread_query_service.coffee
> <https://github.com/loomio/loomio/pull/2369#discussion_r37018107>:
>
> > @@ -52,15 +43,19 @@ angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
> >        parts = options.split ' '
> >        moment().startOf('day').subtract(parseInt(parts[0]), parts[1])
> >
> > -    createFilter = (filter = 'show_all') ->
> > -      (viewData) ->
> > -        _.filter viewData, (thread) ->
> > -          return false if thread.isMuted() and filter != 'show_muted'
> > -          return false if thread.readerNotLoaded()
> > -          switch filter
> > -            when 'show_all'           then true
> > -            when 'show_muted'         then thread.isMuted()
> > -            when 'show_unread'        then thread.isUnread()
> > -            when 'show_participating' then thread.isParticipating()
> > -            when 'show_starred'       then thread.isStarred()
> > -            when 'show_proposals'     then thread.hasActiveProposal()
> > +    applyFilters = (view, filters, queryType) ->
>
> I agree. Maybe we should be merging readers into discussion records and
> just having one object on the client?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r37018107>.
>